networks/create-elb/create-elb.yml

```diff
@@ -113,61 +148,435 @@ Parameters:
+    Type: String
+    AllowedPattern: "((https\\:\\/\\/).+)?"
+  OidcTokenEndpoint:
+    Description: "The token endpoint of the IdP. This must be a full URL, including the HTTPS protocol, the domain, and the path."
+    Type: String
+    AllowedPattern: "((https\\:\\/\\/).+)?"
+  OidcUserInfoEndpoint:
+    Description: "The user info endpoint of the IdP. This must be a full URL, including the HTTPS protocol, the domain, and the path."
+    Type: String
+    AllowedPattern: "((https\\:\\/\\/).+)?"
+
+  # ---------------------------- Parameters - ELB Listener Settings for Cognito Authenticationngs for Cognito Authentication
+  CognitoArn:
+    Description: "The Amazon Resource Name (ARN) of the Amazon Cognito user pool."
+    Type: String
+  CognitoClientId:
+    Description: "The ID of the Amazon Cognito user pool client."
+    Type: String
+  CognitoUserPoolDomain:
+    Description: "The domain prefix or fully-qualified domain name of the Amazon Cognito user pool."
+    Type: String
+
+  # ---------------------------- Parameters - ELB Listener Settings for Redirect
+  RedirectHost:
+    Description: Redirect Action Target Host
+    Type: String
 
 #---------------------------- Condition
 Conditions:
-  ListenerTypeCertificate: !Or [!Equals [HTTPS, !Ref ListenProtocol],!Equals [TLS, !Ref ListenProtocol]]
+  LoadBalancerTypeIsApplication: !Equals ["application", !Ref ElbType]
+  LoadBalancerTypeIsNetwork: !Equals ["network", !Ref ElbType]
+  EnableAccesslLog: !Not [!Equals ["", !Ref ElbAccessLogBucket]]
+  AuthenticateIsForward: !Equals ["forward", !Ref ListenerType]
+  AuthenticateIsRedirect: !Equals ["redirect", !Ref ListenerType]
+  AuthenticateIsFixed: !Equals ["fixed-response", !Ref ListenerType]
+  AuthenticateIsOIDC: !Equals ["OpenID Connect", !Ref ListenerAuthentication]
+  AuthenticateIsCognito: !Equals ["Cognito", !Ref ListenerAuthentication]
+  TargetProtocolTypeWeb: !Or [!Equals [HTTPS, !Ref TargetProtocol], !Equals [HTTP, !Ref TargetProtocol]]
+  ListenerTypeTls: !Or [!Equals [HTTPS, !Ref ListenProtocol], !Equals [TLS, !Ref ListenProtocol]]
+  #UsingCertificate: !Not [!Equals ["Not use certificate", !Ref UseCertificate]]
+  CreateCertificate: !Equals ["Create new certificate from ACM", !Ref UseCertificate]
+
+#---------------------------- Mappings
+Mappings:
+  HealthCheckValues:
+    ip:
+      interval: 30
+      timeout: 5
+      threshold: 5
+      unhealthy: 2
+    instance:
+      interval: 30
+      timeout: 5
+      threshold: 5
+      unhealthy: 2
+    lambda:
+      interval: 35
+      timeout: 30
+      threshold: 5
+      unhealthy: 2
+  ListenerSecurity:
+    Policy:
+      latest: "ELBSecurityPolicy-TLS-1-2-Ext-2018-06"
+  ListenerConfig:
+    Auth:
+      SessionCookie: "ELBAuthCookie"
+      SessionTimeout: 28800
+      OIDCScope: "openid profile email"
+
+# ---------------------------- Metadata
```

Comment:
Could you move this `Metadata` section at least above `Conditions` section?
`Metadata` & `Parameters` sections must be located above `Conditions` & `Resources` sections to enable default value / metadata customization feature.